src/impl_/trampoline.rs

```diff
@@ -201,3 +219,30 @@ where
     py_err.restore(py);
     R::ERR_VALUE
 }
+
+/// Implementation of trampoline for functions which can't return an error.
+///
+/// Panics during execution are trapped so that they don't propagate through any
+/// outer FFI boundary.
+///
+/// Exceptions produced are sent to `sys.unraisablehook`.
+///
+/// # Safety
+///
+/// ctx must be either a valid ffi::PyObject or NULL
+unsafe fn trampoline_inner_void<F>(body: F, ctx: *mut ffi::PyObject)
+where
+    F: for<'py> FnOnce(Python<'py>) -> PyResult<()> + UnwindSafe,
+{
+    let trap = PanicTrap::new("uncaught panic at ffi boundary");
+    let pool = unsafe { GILPool::new() };
+    let py = pool.python();
+    if let Err(py_err) = match panic::catch_unwind(move || body(py)) {
```

Comment:
Isn't this `.map_err(|payload| PanicException::from_panic_payload(payload))`?

Reply:
Actually the type coming from catch unwind is something like `Result<Result<(), PyErr>, PanicPayload>`. I think `.unwrap_or_else()` is what I'm looking for.